pyspark/bigdl/keras/converter.py

```diff
@@ -355,108 +375,135 @@ def __check_is_share_weights(self, kclayer):
 
-    def create_sequential(self, klayer, kclayer):
-        return DefinitionLoader.from_kmodel(klayer)
+    def create_sequential(self):
+        return DefinitionLoader.from_kmodel(self.klayer)
 
-    def create_inputlayer(self, klayer, kclayer):
+    def create_inputlayer(self):
         return BLayer.Identity()
 
-    def create_dense(self, klayer, kclayer):
-        config = kclayer["config"]
+    def create_dense(self):
         # Multiple inputs should share the same input_dim for Dense layer
         # We don't need to respect the tensor index for method `get_input_shape_at`
         # which is internal implementation and `get_input_shape_at` has hided that for us,
         # What we need to use is the input index, not node index, not tensor index.
-        input_shape = klayer.get_input_shape_at(0)
         blayer = BLayer.Linear(
-            input_size=int(input_shape[1]),
-            output_size=config["output_dim"],
-            with_bias=config["bias"],
-            wRegularizer=self.to_bigdl_reg(config["W_regularizer"]),
-            bRegularizer=self.to_bigdl_reg(config["b_regularizer"])
+            input_size=int(self.input_shape[-1]), # last dim is input_dim
+            output_size=self.config["output_dim"],
+            with_bias=self.config["bias"],
+            wRegularizer=self.to_bigdl_reg(self.config["W_regularizer"]),
+            bRegularizer=self.to_bigdl_reg(self.config["b_regularizer"])
         )
-        return self.combo_parameter_layer(blayer, config)
+        return self.combo_parameter_layer(blayer, self.config)
 
-    def create_timedistributeddense(self, klayer, kclayer):
-        config = kclayer["config"]
-        input_shape = klayer.get_input_shape_at(0)
+    def create_timedistributeddense(self):
         blayer = BLayer.TimeDistributed(BLayer.Linear(
-            input_size=int(input_shape[2]),
-            output_size=config["output_dim"],
-            with_bias=config["bias"],
-            wRegularizer=self.to_bigdl_reg(config["W_regularizer"]),
-            bRegularizer=self.to_bigdl_reg(config["b_regularizer"])
+            input_size=int(self.input_shape[-1]),
+            output_size=self.config["output_dim"],
+            with_bias=self.config["bias"],
+            wRegularizer=self.to_bigdl_reg(self.config["W_regularizer"]),
+            bRegularizer=self.to_bigdl_reg(self.config["b_regularizer"])
         ))
-        return self.combo_parameter_layer(blayer, config)
+        return self.combo_parameter_layer(blayer, self.config)
+
+    def create_timedistributed(self):
+        # remove the time dimension in shape (batch, time, other dims)
+        inner_input_shape = (self.input_shape[0], ) + self.input_shape[2:]
+        blayer = LayerConverter(self.klayer.layer, self.config['layer'], inner_input_shape).create()
+        return BLayer.TimeDistributed(blayer)
+
+    def create_bidirectional(self):
+        klayer_forward = self.klayer.layer
```

Comment:
can we use `BiRecurrent` directly?

Reply:
I previously used `BiRecurrent` directly, but the problem is that `BiRecurrent` doesn't support `return_sequences=False` just like other recurrent layers (for other recurrent layers, we use `Select` layer if `return_sequences=False`. For bidirectional, it is not trivial to select directly, so I implement by myself.